### PR TITLE
add richestCustomerWealth page

### PR DESCRIPTION
**Pull request by raveenreddy0**
PR to add richestCustomerWealth.tsx and linesOfCode.ts in /api/coding/ folder to store the leetcode solution in a list of objects.

Some notes:
- page currently displays the leetcode question [prompt](https://leetcode.com/problems/richest-customer-wealth/), my solution, next / previous buttons, inputs for leetcode question, and variables in solution
- idea behind next / previous buttons is to iterate through solution step by step and highlight each line while visualizing how the variables change while working through the solution
- the solution is rendered through a map of array `linesOfCode` but I can't figure out how to highlight each line depending on the state variable `lineNum`
- currently all lines of code are highlighted when `lineNum` = 6; 6 being a placeholder for where the current `lineNum` would be referenced
- forgot to create a new branch so the old js question commits are still included in this PR whoops 

Here is what it looks so far:
Screen Shot 2022-02-11 at 1 44 55 PM

### CODE REVIEW

File: pages/api/coding/LinesOfCode.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//         console.log('the final account sum for account: ' + index + "is" + account_sum)
+//         balances.push(account_sum)
+//     })
+    
+//     console.log("the account balances are:" + balances)
+    
+//     // sort list descending
+//     balances.sort((a, b) => {return b-a})
+    
+//     return balances[0]
```
Comment (LavaBoy202) [inline pages/api/coding/LinesOfCode.ts:81]: Don't commit commented out code

File: components/coding/JSQuiz.tsx [hunk 2]
```diff
@@ -58,6 +58,51 @@ const JSQuiz = ({}: JSQuizProp) => {
       },
       answer: "variable x;",
     },
+    {
+      displayQuestion: "Which of the following expressions evaluates as True",
```
Comment (LavaBoy202) [inline components/coding/JSQuiz.tsx:62]: It's good to note all your changes in a PR. Add a bullet point in your PR description saying that you added some more questions to the JS quiz.

**Review by LavaBoy202 — COMMENTED** [review]
Nice job! It's pretty clean code, just couple of small improvements here and there. Make sure the vercel tests pass before merging.

File: pages/api/coding/LinesOfCode.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+export const linesOfCode = [
+    {
+        line: 1,
```
Comment (vithushan19) [inline pages/api/coding/LinesOfCode.ts:3]: This can be array of strings where you use the index

File: pages/coding/richestCustomerWealth.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          </p>
+        </div>
+      </div>
+      <div className="grid grid-cols-2 items-center justify-center gap-4 heropattern-bamboo-gray-300 pt-8">
+        <div className="flex flex-col gap-4 p-4 bg-white shadow-lg rounded-xl">
+          <p className="font-bold text  border-b-2 border-black">
+            Code to Evaluate
+          </p>
+
+          {linesOfCode.map(({ line, text }) => {
```
Comment (vithushan19) [inline pages/coding/richestCustomerWealth.tsx:34]: 🔥

**Review by vithushan19 — COMMENTED** [review]
Looking good!

**Comment by raveenreddy0** [comment]
@vithushan19 @LavaBoy202 

Realized I've never closed the original PR and I'm getting kind of lost in this branch so sorry for that - will try to close this one out and be better about that going forward. I did try to add commits as I go so hopefully that makes it easier to follow. 

The new commits here iterate / highlight through the solution more dynamically based on Lavan's suggestions 

Some highlights / issues:
- added several new state variables to keep track of the solution steps and variables as we iterate line by line
- added Reset button that sets all state variables to initial values onClick
- created an iterateOnSolution function called by Next Button's onClick which has conditional logic to iterate through both loops appropriately - i'm guessing there's a more elegant / efficient way to work through this so can work on refactoring in next iterations
- added some code to highlight each element of the Accounts array as it's iterated through 
- i'm trying to sort through the balances array once it's populated, and pass it through setBalancesState but this code doesn't appear to be working
- in general looking for a cleaner way to render the arrays.. for the balances arrays, I created a renderAccounts object bu this seems kinda unclean??

**Review by vithushan19 — APPROVED** [review]

**Pull request merged**